tools/install.sh

```diff
@@ -155,10 +155,15 @@ chmod 770 ../smarty/templates_c
 
 # Changing group to 'www-data' or 'apache' to give permission to create directories in Document Repository module
 # Detecting distribution
-os_distro=$(hostnamectl |awk -F: '/Operating System:/{print $2}'|cut -f2 -d ' ')
+if ! os_distro=$(hostnamectl |awk -F: '/Operating System:/{print $2}'|cut -f2 -d ' ')  2> /dev/null
```

Comment:
I think a better way to do this is:

```bash
command -v hostnamectl > /dev/null
if [[ "$?" -eq "0" ]]; then
     os_distro=$(hostnamectl |awk -F: '/Operating System:/{print $2}'|cut -f2 -d ' ')
else
     os_distro='other'
fi
```

`command -v hostnamectl` checks for the existence of the program `hostnamectl`.

If it exists, the return code of this statement will be `0`; otherwise, it will be `1`.

The variable `$?` captures the exit code of the last command run.

Then the if statement checks whether `hostnamectl` was found. If so, we are on some kind of Linux that we support and can parse the output of hostnamectl. Otherwise we default to other.

`command` is standard and exists on macOS so this should cover your issue.

Reply:
Hey @johnsaigle I tested your code just now and nothing gets outputted after that line for the rest of the installation script and contrary to my fix.

Reply:
That's really weird but I also confirmed that on macOS. `command -v ...` works fine on Ubuntu but fails silently in the same script on macOS. No idea why, couldn't find an answer via Google.